### ISSUE TITLE
Return wrong type when using `SELECT SUM()`

### ISSUE THREAD

**Issue by teric1024** [sqlc-dev/sqlc#1106]
In the sqlc playground, I edit the `query.sql` as followed
```sql
-- Example queries for sqlc
CREATE TABLE orders (
  id   BIGSERIAL PRIMARY KEY,
  quantity decimal      NOT NULL,
  order_catalog  int
);

-- name: SumOrder :one
SELECT SUM(quantity) FROM orders;
```
Then, `db/query.sql.go` would be
```go
// Code generated by sqlc. DO NOT EDIT.
// source: query.sql

package db

import (
	"context"
)

const sumOrder = `-- name: SumOrder :one
SELECT SUM(quantity) FROM orders
`

func (q *Queries) SumOrder(ctx context.Context) (int64, error) {
	row := q.db.QueryRowContext(ctx, sumOrder)
	var sum int64
	err := row.Scan(&sum)
	return sum, err
}
```
I think the return type of SumOrder should not be int64. It should be string or something else.

**Comment by gnuletik** [comment]
Still an issue, also when using `float` type:

https://play.sqlc.dev/p/fc1b90f3e13264a46d15bbb4ef3867de28b0bf6839d6fb10325d65407b61aebe

**Comment by kyleconroy** [comment]
This is fixed in [v1.23.0](https://docs.sqlc.dev/en/latest/reference/changelog.html#v1-23-0) by enabling the [database-backed query analyzer](https://docs.sqlc.dev/en/latest/howto/generate.html#enhanced-analysis-with-managed-databases). We added a [test case](https://github.com/sqlc-dev/sqlc/tree/main/internal/endtoend/testdata/bsum_type/postgresql/pgx) for this issue so it won’t break in the future.

You can play around with the [working example on the playground](https://play.sqlc.dev/p/1f2b593e7523980cf1c19d4a1f3bc15194c9b859323f13102c0eb2568186b623)
